contrib/lib/src/space_helmet/policy.rs

```diff
@@ -0,0 +1,169 @@
+pub enum ExpectCTPolicy<'a> {
+    Enforce(MaxAge),
+    Report(MaxAge, Uri<'a>),
+    ReportAndEnforce(MaxAge, Uri<'a>),
+}
+
+impl<'a> Default for ExpectCTPolicy<'a> {
+    fn default() -> ExpectCTPolicy<'a> {
+        ExpectCTPolicy::Enforce(EXPECTCT_MAX_AGE_DEFAULT)
+    }
+}
+
+impl<'a, 'b> From<&'a ExpectCTPolicy<'a>> for Header<'b> {
+    fn from(policy: &ExpectCTPolicy<'a>) -> Header<'b> {
+        let policy_string = match policy {
+            ExpectCTPolicy::Enforce(max_age) => format!("max-age={}, enforce", max_age),
+            ExpectCTPolicy::Report(max_age, url) => {
+                format!("max-age={}, report-uri=\"{}\"", max_age, url)
+            }
+            ExpectCTPolicy::ReportAndEnforce(max_age, url) => {
+                format!("max-age={}, enforce, report-uri=\"{}\"", max_age, url)
+            }
+        };
+        Header::new("Expect-CT", policy_string)
+    }
+}
+
+///[X-Content-Type-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options) description.
+pub enum NoSniffPolicy {
+    Enable,
+}
+
+impl Default for NoSniffPolicy {
+    fn default() -> NoSniffPolicy {
+        NoSniffPolicy::Enable
+    }
+}
+
+impl<'a, 'b> From<&'a NoSniffPolicy> for Header<'b> {
+    fn from(_policy: &NoSniffPolicy) -> Header<'b> {
+        Header::new("X-Content-Type-Options", "nosniff")
+    }
+}
+
+///[Strict-Transport-Security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) description.
+pub enum HSTSPolicy {
+    Enable(MaxAge),
+    IncludeSubDomains(MaxAge),
+    Preload(MaxAge),
+}
+
+//XXX add debug setting.
+///One year in seconds, a setting for production environments.
+pub const HSTS_MAX_AGE_DEFAULT: u32 = 31536000;
+
+impl Default for HSTSPolicy {
+    fn default() -> HSTSPolicy {
+        HSTSPolicy::Enable(HSTS_MAX_AGE_DEFAULT)
+    }
+}
+
+impl<'a, 'b> From<&'a HSTSPolicy> for Header<'b> {
+    fn from(policy: &HSTSPolicy) -> Header<'b> {
+        let policy_string = match policy {
+            HSTSPolicy::Enable(max_age) => format!("max-age={}", max_age),
+            HSTSPolicy::IncludeSubDomains(max_age) => {
+                format!("max-age={} ; includeSubDomains", max_age)
+            }
+            HSTSPolicy::Preload(max_age) => format!("max-age={} ; preload", max_age),
+        };
+        Header::new("Strict-Transport-Security", policy_string)
+    }
+}
+
+///[X-Frame-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) description.
+pub enum FramePolicy<'a> {
+    SameOrigin,
+    Deny,
+    AllowFrom(Uri<'a>),
+}
+
+impl<'a> Default for FramePolicy<'a> {
+    fn default() -> FramePolicy<'a> {
+        FramePolicy::SameOrigin
+    }
+}
+
+impl<'a, 'b> From<&'a FramePolicy<'a>> for Header<'b> {
+    fn from(policy: &FramePolicy<'a>) -> Header<'b> {
+        let policy_string = match policy {
+            FramePolicy::Deny => String::from("DENY"),
```

Comment:
Use `.into()` instead of calling `String::from`: `"DENY".into()`. Better yet, to avoid allocating when it's just a static string, the type of the match arms should be `Cow`. That way, `"DENY"` and `"SAMEORIGIN"` can be a `Cow::Borrowed` and the `format!("ALLOW-FROM {}", uri)` can be a `Cow::Owned`.

This applied to all other similar cases as well. To avoid boiler plate, you can write the code as:

```rust
let policy_string: Cow<'static, str> = match policy {
    FramePolicy::Deny => "DENY".into(),
    FramePolicy::SameOrigin => "SAMEORIGIN".into(),
    FramePolicy::AllowFrom(uri) => format!("ALLOW-FROM {}", uri).into()
};

Header::new("X-Frame-Options", policy_string)
```